client-core/src/service/global_state_service.rs

```diff
@@ -98,6 +102,41 @@ where
                     .map(|global_state_optional| global_state_optional.unwrap_or_default())
             })
     }
+
+    /// load trust state
+    pub fn load_trust_state(&self) -> Result<Option<lite::TrustedState>> {
+        // get current trusted state
+        let opt = self.storage.get(KEYSPACE, TRUST_STATE_KEY)?;
+        if let Some(bytes) = opt {
+            let state = serde_json::from_slice(bytes.as_slice()).chain(|| {
```

Comment:
AFAIK the rest of the client stores data as binary using SCALE codec, but that's quite chain-specific and it'd be tiresome to retrofit all the Tendermint data types (as we plan to depend on the mainstream rather than a custom fork), so I think it's ok to store this data in json -- @devashishdxt @leejw51 @calvinlauco any thoughts?

Reply:
We can also use `bincode` which'll be more performant. `json` is also fine.

Reply:
json is fine

Reply:
Since we are going to depend on the mainstream, I also think using JSON seems better in the long term; When later we have more stress test result, we can have more data to determine whether we need to change this approach.